src/modules/mavlink/streams/ACTUATOR_OUTPUT_STATUS.hpp

```diff
@@ -0,0 +1,107 @@
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#ifndef ACTUATOR_OUTPUT_STATUS_HPP
+#define ACTUATOR_OUTPUT_STATUS_HPP
+
+class MavlinkStreamActuatorOutputStatus : public MavlinkStream
+{
+public:
+	const char *get_name() const override
+	{
+		return MavlinkStreamActuatorOutputStatus::get_name_static();
+	}
+
+	static constexpr const char *get_name_static()
+	{
+		return "ACTUATOR_OUTPUT_STATUS";
+	}
+
+	static constexpr uint16_t get_id_static()
+	{
+		return MAVLINK_MSG_ID_ACTUATOR_OUTPUT_STATUS;
+	}
+
+	uint16_t get_id() override
+	{
+		return get_id_static();
+	}
+
+	static MavlinkStream *new_instance(Mavlink *mavlink)
+	{
+		return new MavlinkStreamActuatorOutputStatus(mavlink);
+	}
+
+	unsigned get_size() override
+	{
+		return (_act_output_sub.advertised()) ? (MAVLINK_MSG_ID_ACTUATOR_OUTPUT_STATUS_LEN + MAVLINK_NUM_NON_PAYLOAD_BYTES) : 0;
+	}
+
+private:
+	uORB::Subscription _act_output_sub{ORB_ID(actuator_outputs)};
+
+	/* do not allow top copying this class */
+	MavlinkStreamActuatorOutputStatus(MavlinkStreamActuatorOutputStatus &) = delete;
+	MavlinkStreamActuatorOutputStatus &operator = (const MavlinkStreamActuatorOutputStatus &) = delete;
+
+protected:
+	explicit MavlinkStreamActuatorOutputStatus(Mavlink *mavlink) : MavlinkStream(mavlink)
+	{}
+
+	bool send() override
+	{
+		actuator_outputs_s act;
+
+		if (_act_output_sub.update(&act)) {
+			mavlink_actuator_output_status_t msg{};
+
+			msg.time_usec = act.timestamp;
+			msg.active = act.noutputs;
+
+			static constexpr size_t act_output_size = sizeof(act.output) / sizeof(act.output[0]);
+			static constexpr size_t msg_actuator_size = sizeof(msg.actuator) / sizeof(msg.actuator[0]);
+
+			for (unsigned i = 0; (i < act_output_size) && (i < msg_actuator_size); i++) {
```

Comment:
I think this is what you'd want?

```suggestion
			for (unsigned i = 0; i < math::min(msg_actuator_size, act.noutputs); i++) {
```